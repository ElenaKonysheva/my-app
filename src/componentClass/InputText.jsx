import React, { Component } from 'react';

export class InputText extends Component {
  render() {
    return (
      <input
        type="text"
        onChange={this.props.changeText}
        value={this.props.value}
        placeholder="введите ваше сообщение"
      />
    );
  }
}
