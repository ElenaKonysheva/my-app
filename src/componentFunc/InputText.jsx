import React from 'react';

export const InputText = (props) => {
    return <input type='text' onChange={props.changeText} value={props.value}
        placeholder='введите ваше сообщение' />
}


