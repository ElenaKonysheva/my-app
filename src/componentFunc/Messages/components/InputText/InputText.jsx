import React from 'react';
export const InputText = (props) => {
  return (
    <textarea data-testid="custom-element"
      placeholder="write your message"
      onChange={props.changeText}
      value={props.value}
    ></textarea>
  );
};
