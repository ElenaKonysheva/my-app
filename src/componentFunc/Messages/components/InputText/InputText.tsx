import React from 'react';
import Box from '@mui/material/Box';
import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import AccountCircle from '@mui/icons-material/AccountCircle';
// import PropTypes from 'prop-types';
// interface InputTextProps {
//   value: string;
//   changeText: () => void;
// }

export const InputText = (props: {
  changeText:
    | React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
    | (() => void);
  value: string;
}) => {
  return (
    <Box sx={{ '& > :not(style)': { m: 4 } }}>
      <TextField
        data-testid="custom-element"
        placeholder={'write your message'}
        onChange={props.changeText}
        value={props.value}
        id="input-with-icon-textfield"
        label="TextField"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <AccountCircle />
            </InputAdornment>
          ),
        }}
        variant="standard"
      />
    </Box>
  );
};
// InputText.propTypes = {
//   changeText: PropTypes.func,
//   value: PropTypes.string,
// };
