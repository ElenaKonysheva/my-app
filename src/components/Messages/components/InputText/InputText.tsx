import React, { FC } from 'react';
import Box from '@mui/material/Box';
import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import AccountCircle from '@mui/icons-material/AccountCircle';
interface InputTextProps {
  changeText?:
    | React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
    | (() => void);
  value?: string;
}
export const InputText: FC<InputTextProps> = (props) => {
  return (
    <Box sx={{ '& > :not(style)': { m: 4 } }}>
      <TextField
        data-testid="custom-element"
        placeholder={'write your message'}
        onChange={props.changeText}
        value={props.value}
        id="input-with-icon-textfield"
        label="TextField"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <AccountCircle />
            </InputAdornment>
          ),
        }}
        variant="standard"
      />
    </Box>
  );
};
