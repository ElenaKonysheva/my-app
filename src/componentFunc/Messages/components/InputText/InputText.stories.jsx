import React from 'react';

import { InputText } from './InputText';

export default {
  title: 'MyComponents/InputText',
  component: InputText,
  argTypes: {
    changeText: { action: 'change' },
  },
};

const Template = (args) => <InputText {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  placeholder: 'write your message',
  label: 'InputText',
};

export const Secondary = Template.bind({});
Secondary.args = {
  placeholder: '',
  label: 'InputText',
};
