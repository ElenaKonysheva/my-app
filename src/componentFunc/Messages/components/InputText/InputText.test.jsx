import React from "react";
import { InputText } from "./InputText";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@babel/plugin-transform-runtime"

describe("InputText", () => {
  it("render component", () => {
    render(<InputText />);
  });
  it("render with snapshot", () => {
    const { asFragment } = render(<InputText />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("render with placeholder", () => {
    render(<InputText />);
    expect(
      screen.getByPlaceholderText(/write your message/i)
    ).toBeInTheDocument();
  });
  it("render myltiply components", () => {
    render(<InputText />);
    expect(screen.getByTestId("custom-element")).toBeInTheDocument();
  });
  it("input submit return number", () => {
    const setup = () => {
      const utils = render(<InputText />);
      const input = utils.getByTestId("custom-element");
      return {
        input,
        ...utils,
      };
    };
    const { input } = setup();
    fireEvent.change(input, { target: { value: "23" } });
    expect(input.value).toBe("23");
  });
  it("input submit return string", () => {
    const setup = () => {
      const utils = render(<InputText />);
      const input = utils.getByTestId("custom-element");
      return {
        input,
        ...utils,
      };
    };
    const { input } = setup();
    fireEvent.change(input, { target: { value: "привет" } });
    expect(input.value).toBe("привет");
  });
  it("input submit return null", () => {
    const setup = () => {
      const utils = render(<InputText />);
      const input = utils.getByTestId("custom-element");
      return {
        input,
        ...utils,
      };
    };
    const { input } = setup();
    fireEvent.change(input, { target: { value: "" } });
    expect(input.value).toBe("");
  });

});
