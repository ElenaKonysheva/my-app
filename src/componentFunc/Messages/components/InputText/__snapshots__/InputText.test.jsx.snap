// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText render with snapshot 1`] = `
<DocumentFragment>
  <textarea
    data-testid="custom-element"
    placeholder="write your message"
  />
</DocumentFragment>
`;
